orchestrator/cosmos_gravity/src/send.rs

```diff
@@ -182,8 +182,8 @@ pub async fn send_messages<CS: CosmosSigner>(
     args.fee.gas_limit = cmp::max(gas_limit as u64, 500000 * messages.len() as u64);
 
     // compute the fee as fee=ceil(gas_limit * gas_price)
-    let fee_amount: f64 = args.fee.gas_limit as f64 * gas_price.0;
-    let fee_amount: u64 = fee_amount.abs().ceil() as u64;
+    let fee_amount = (args.fee.gas_limit as u128).checked_mul(gas_price.0 as u128)
```

Comment:
not sure it is the best way to fix it
I think it only work fine for cronos as our gas price is not float number which is `5000000000000`basecro
for crypto.org chain, the gas price is `0.025`basecro

Reply:
the gas price should be in `basetcro` no? so technically a positive integer?

Reply:
Is `u128` enough? For solo machine we use [`U256`](https://docs.rs/primitive-types/latest/primitive_types/struct.U256.html).

Reply:
I'm not sure if there are integration tests for this. But can you add a unit test to check it this is enough?

Reply:
we have tested on testnet with current value and it seems fine

max gas per block is 20000000 and CRO has 18 decimals, max supply 30 billions

The potential max should be around 6 * 10^35 which is still below max u128 so it may be okay?